helpers/getImageSrcset.js

```diff
@@ -0,0 +1,49 @@
+'use strict';
+
+const factory = globals => {
+    return function(image, presetName, defaultImageUrl) {
+        // Regex to test size string is of the form 123x123
+        var sizeRegex = /^(\d+?)x(\d+?)$/;
+        // Regex to test to that srcset descriptor is of the form 1x 1.5x 2x OR 123w
+        var descriptorRegex = /(^\d+w$)|(^([0-9](\.[0-9]+)?)x)$/;
+        // Get theme settings from context
+        var settings = globals.getThemeSettings() || {};
+        var srcsets = settings[presetName] || [];
```

Comment:
To my knowledge, we don't support arrays or nested objects in the theme configuration right now -- just plain key/value pairs.

Reply:
Moved to arguments per your suggestion.